Resources/Private/JavaScript/asset-upload-screen/src/AssetUploadScreen.tsx

```diff
@@ -0,0 +1,153 @@
+import React, { createRef } from 'react';
+import { connect } from 'react-redux';
+import { ApolloClient, ApolloLink } from '@apollo/client';
+import { createUploadLink } from 'apollo-upload-client';
+
+// Neos dependencies are provided by the UI
+// @ts-ignore
+import { neos } from '@neos-project/neos-ui-decorators';
+// @ts-ignore
+import { actions } from '@neos-project/neos-ui-redux-store';
+
+import NewAssetUpload from './NewAssetUpload';
+
+import { MediaUiProvider, typeDefs as TYPE_DEFS_CORE } from '@media-ui/core';
+import MediaApplicationWrapper from '@media-ui/core/src/components/MediaApplicationWrapper';
+import { CacheFactory, createErrorHandler } from '@media-ui/media-module/src/core';
+import { typeDefs as TYPE_DEFS_ASSET_USAGE } from '@media-ui/feature-asset-usage';
+
+let apolloClient = null;
+
+interface AssetUploadScreenProps {
+    i18nRegistry: I18nRegistry;
+    frontendConfiguration: FeatureFlags;
+    neos: Record<string, unknown>;
+    type: AssetType | 'images'; // The image editor sets the type to 'images'
+    onComplete: (result: { object: { __identity: string } }) => void;
+    isLeftSideBarHidden: boolean;
+    isNodeCreationDialogOpen: boolean;
+    toggleSidebar: () => void;
+    addFlashMessage: (title: string, message: string, severity?: string, timeout?: number) => void;
+    constraints?: SelectionConstraints;
+}
+
+interface AssetUploadScreenState {
+    initialLeftSideBarHiddenState: boolean;
+    initialNodeCreationDialogOpenState: boolean;
+}
+
+export class AssetUploadScreen extends React.PureComponent<AssetUploadScreenProps, AssetUploadScreenState> {
```

Comment:
Do I See correctly that this plugin will not do anything until Neos.UI has been extended with the additional feature?

If that is the case, I would prefer to have this PR split and have this screen separately. Or is there a good reason against that?

Reply:
Yes, you are correct. The plugin will not do anything, without the adjustments in the Neos.Ui . However, in my opinion it is not so easy to split this PR. There are changes in other components such as the preview component that are indirectly needed for this plugin. In my view, a potential split PR would therefore have to be based on this PR structure and cannot be developed for itself against the main branch. In my view, this would only add unnecessary complexity. However, we can look at the changes together again. Perhaps we can find a solution that makes sense for both of us.

Reply:
My problem is, that I cannot test the dialog in my normal environment for this package and therefore changes might break it without notice.

Reply:
I wonder if we could add the dialog in a way that would work right now 🤔

Reply:
I think this topic is now resolved, because my changes were merged into the Neos 8.4 branch. https://github.com/neos/neos-ui/pull/3302